cpp/tests/transform/row_bit_count_test.cu

```diff
@@ -389,14 +393,34 @@ build_nested_and_expected_column(std::vector<bool> const& struct_validity)
   auto struct_column =
     cudf::test::structs_column_wrapper({list, ages_column, i16_column}, struct_validity);
 
+  // Each struct has size:
+  //    (1 offset (4 bytes) + (list size if row is valid) + 1 validity bit) +
+  //    (1 float + 1 validity bit) +
+  //    (1 int16_t + 1 validity bit) +
+  //    (1 validity bit)
+  auto const struct_size_iter =
+    cudf::detail::make_counting_transform_iterator(0, [&](auto const idx) {
+      return (CHAR_BIT * (sizeof(cudf::offset_type) +
+                          sizeof(cudf::size_type) * (struct_validity[idx] ? list_sizes[idx] : 0)) +
+              1) +
+             (sizeof(float) * CHAR_BIT + 1) + (sizeof(int16_t) * CHAR_BIT + 1) + 1;
+    });
+
   // wrap in a list
   std::vector<int> outer_offsets{0, 1, 1, 3, 6, 7, 8};
   cudf::test::fixed_width_column_wrapper<int> outer_offsets_col(outer_offsets.begin(),
                                                                 outer_offsets.end());
   auto const size = static_cast<cudf::column_view>(outer_offsets_col).size() - 1;
 
-  cudf::test::fixed_width_column_wrapper<cudf::size_type> expected_sizes{
-    276, 32, 520, 572, 212, 212};
```

Comment:
Could this be resurrected and not perpetuate the problem of keeping the code logic matching here?

Reply:
These numbers are computed using the expected behavior of the API. Previously these numbers were computed with non-empty nulls (which was the expected behavior), now they are recomputed based on empty nulls (new expected behavior, which is believed to be correct). In the future, if we have new breaking changes (such as changing the offset type from int32 to int64, although it is very unlikely) then this could be resurrected again.

Reply:
This appears to be now replaced with a recoding of the expected behavior instead of ground-truth results. Would prefer checking against ground-truth.

Reply:
I disagree. We should prefer to check against ground-truth instead of recoding the internal logic in the unit test.